### PR TITLE
[FLINK-35884][mysql-pipeline-connector] MySQL pipeline support snapshot chunk key-column

### PR DESCRIPTION
**Pull request by beryllw**
Flink-connector-mysql-cdc `MySqlSourceOptions` supports specifying the parameter scan.incremental.snapshot.chunk.key-column to divide chunks, pipeline connector should also support.

### CODE REVIEW

**Comment by beryllw** [comment]
@ruanhang1993 PTAL

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java [hunk 1]
```diff
@@ -243,6 +271,14 @@ private static List<String> getTableList(MySqlSourceConfig sourceConfig, Selecto
                 .collect(Collectors.toList());
     }
 
+    private static List<ObjectPath> getChunkKeyColumnTableList(
+            MySqlSourceConfig sourceConfig, Selectors selectors) {
+        return MySqlSchemaUtils.listTables(sourceConfig, null).stream()
```
Comment (ruanhang1993) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java:276]: Could we invoke `listTables` once and filter tables in the same list for each selector?

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java [hunk 2]
```diff
@@ -181,6 +184,30 @@ public DataSource createDataSource(Context context) {
...
+                    Selectors chunkKeySelector =
+                            new Selectors.SelectorsBuilder().includeTables(splits[0]).build();
+                    List<ObjectPath> tableList =
+                            getChunkKeyColumnTableList(
+                                    configFactory.createConfig(0), chunkKeySelector);
+                    for (ObjectPath table : tableList) {
+                        chunkKeyColumnMap.put(table, splits[1]);
+                    }
+                } else {
+                    throw new IllegalArgumentException(
```
Comment (ruanhang1993) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java:202]:
```suggestion
                    throw new IllegalArgumentException(SCAN_INCREMENTAL_SNAPSHOT_CHUNK_KEY_COLUMN.key() + " = " + chunkKeyColumns + " failed to be parsed in this part '" + chunkKeyColumn + "'.");
```

**Comment by beryllw** [comment]
@ruanhang1993 PTAL, thanks.

**Comment by beryllw** [comment]
[compile_and_test](https://github.com/apache/flink-cdc/actions/runs/10195953793/job/28335622434?pr=3490#logs)
```
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 122.202 s <<< FAILURE! - in org.apache.flink.cdc.connectors.starrocks.sink.StarRocksMetadataApplierITCase
Error:  org.apache.flink.cdc.connectors.starrocks.sink.StarRocksMetadataApplierITCase  Time elapsed: 122.202 s  <<< ERROR!
java.util.concurrent.CompletionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image starrocks/allin1-ubuntu:3.2.6
```

[data_stream_migration_test](https://github.com/apache/flink-cdc/actions/runs/10195953793/job/28335732338?pr=3490#logs) and [pipeline_migration_test](https://github.com/apache/flink-cdc/actions/runs/10195953793/job/28335732490?pr=3490#logs)
```
line 1: docker-compose: command not found
```

**Comment by beryllw** [comment]
@leonardBang PTAL

**Review by leonardBang — APPROVED** [review]
Thanks @beryllw for the contribution and @ruanhang1993 for the review, LGTM, wait the CI green

**Comment by beryllw** [comment]
```
2024-08-12T03:56:34.3999892Z [WARNING] Tests run: 20, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 624.39 s - in org.apache.flink.cdc.connectors.oracle.table.OracleConnectorITCase
2024-08-12T03:56:34.4428837Z [INFO] 
2024-08-12T03:56:34.4429524Z [INFO] Results:
2024-08-12T03:56:34.4430148Z [INFO] 
2024-08-12T03:56:34.4432410Z [ERROR] Failures: 
2024-08-12T03:56:34.4436613Z [ERROR]   NewlyAddedTableITCase.testJobManagerFailoverForNewlyAddedTable:229->testNewlyAddedTableOneByOne:585->testNewlyAddedTableOneByOne:725->OracleSourceTestBase.assertEqualsInAnyOrder:111->OracleSourceTestBase.assertEqualsInOrder:119 arrays first differed at element [0]; expected:<...16874195632735147, C[HINA], BEIJING, BEIJING W...> but was:<...16874195632735147, C[hina], BEIJING, BEIJING W...>
2024-08-12T03:56:34.4440159Z [INFO] 
2024-08-12T03:56:34.4440793Z [ERROR] Tests run: 56, Failures: 1, Errors: 0, Skipped: 0
```

**Pull request merged**